core/common/src/main/java/alluxio/security/authentication/DefaultAuthenticationServer.java

```diff
@@ -178,7 +163,7 @@ private void cleanupStaleClients() {
     // Unregister stale clients.
     LOG.debug("Found {} stale channels for cleanup.", staleChannels.size());
     for (UUID clientId : staleChannels) {
-      unregisterChannel(clientId);
+      mChannels.remove(clientId).getSaslServerDriver().close();
```

Comment:
why does this not use `unregisterChannel`?

Reply:
because the `AuthenticatedChannelServerDriver` owns calling unregisterChannel. We assume, correctly, that there is no channel registered here that doesn't have an associated driver. This switch made handling of closing much simpler.